tests/testthat/test_lmimpute.R

```diff
@@ -0,0 +1,69 @@
+library(testthat)
+library(modeldata)
+
+context("linear regression imputation")
+data(ames)
+
+ames_dat <- ames %>%
+  select(Lot_Frontage, Lot_Area, Total_Bsmt_SF) %>%
+  mutate(Lot_Frontage = na_if(Lot_Frontage, 0)) %>%
+  mutate(Total_Bsmt_SF = na_if(Total_Bsmt_SF, 0))
+
+tg_dat <- ToothGrowth %>%
+  mutate(dose = na_if(dose, 0.5))
+
+
+test_that("Does the imputation (no NAs), and does it correctly.", {
```

Comment:
Yes. It seems like a bad idea but using `as.integer()` to truncate the result is better than changing the data type IMO. It would also me consistent with the other imputation methods.